modular/blocksyncer/blocksyncer_options.go

```diff
@@ -374,6 +396,52 @@ func (b *BlockSyncerModular) prepareMasterFlagTable() error {
 	return nil
 }
 
+func (b *BlockSyncerModular) dbStatistics(ctx context.Context) {
+	duration := b.DataStatisticsDuration
+	if duration == 0 {
+		duration = 3600
+	}
+	ticker := time.NewTicker(time.Duration(duration) * time.Second)
+	defer ticker.Stop()
+	var totalCount, sealCount []int64
+	var err error
+	for range ticker.C {
+		latestBlockHeightAny := Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Load()
+		latestBlockHeight := latestBlockHeightAny.(int64)
+		if latestBlockHeight < 10 {
+			continue
+		}
+		if b.BlockResultStorage && latestBlockHeight-b.MaxBlockNum > 0 {
+			if delErr := db.Cast(b.parserCtx.Database).DeleteBlockResult(ctx, latestBlockHeight-b.MaxBlockNum); delErr != nil {
```

Comment:
The process of deleting stale block result data can be separated from block-syncer main thread by using a scheduler?